web/src/server_events_dispatch.js

```diff
@@ -824,6 +825,7 @@ export function dispatch_normal_event(event) {
                             message_details: event.message_details,
                         });
                     }
+                    inbox_ui.update();
```

Comment:
I think I'd prefer for this sort of update to be called from `process_read_messages_event` / `process_unread_messages_event` instead of here -- the intent here is pretty clearly for those functions to handle all aspects of the live update.

Reply:
Done.

Reply:
Seeing this, I do wonder a bit if we should instead be doing the call from:

```
export function update_unread_counts(skip_animations = false) {                                         
    // Pure computation:                                                                                
    const res = unread.get_counts();                                                                    
                                                                                                        
    // Side effects from here down:                                                                     
    // This updates some DOM elements directly, so try to                                               
    // avoid excessive calls to this.                                                                   
    // See `ui_init.initialize_unread_ui` for the registered hooks.                                     
    for (const hook of update_unread_counts_hooks) {                                                    
        hook(res, skip_animations);                                                                     
    }                                                                                                   
```

Probably we can make a follow-up issue to investigate that; this version should work and that's just for making the code better.

Reply:
Opened #26687